daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Compiler.scala

```diff
@@ -211,6 +206,37 @@ private[lf] final class Compiler(
       case None => expr
     }
 
+  @inline
+  private[this] def app(f: SExpr, a: SExpr) = SEApp(f, Array(a))
+
+  @inline
+  private[this] def let(bound: SExpr)(body: Position => SExpr) =
+    SELet1General(bound, body(nextPosition()))
+
+  @inline
+  private[this] def function(body: Position => SExpr): SExpr =
```

Comment:
would `unaryFunction` be a clearer name?

Reply:
changed.

Reply:
> Looks good, but scary! :)

To be really scared you have to consider this changes together with  #7492, #7502, #7472 
;)